src/org/usfirst/frc4904/robot/commands/AutoShifter.java

```diff
@@ -64,6 +66,16 @@ protected void execute() {
 			shiftDownCommand.start();
 			return;
 		}
+		// If we're flooring it and nothing's in our way and we're going down the field, shiftup.
+		float navxYaw = RobotMap.Component.navx.getYaw();
+		boolean isGoingDownTheField = Math.abs(navxYaw) <= 30;
```

Comment:
@NikhilSuresh24 thanks. The only reason I haven't tested is because time with the robot is limited.